### ISSUE TITLE
promise api?

### ISSUE THREAD

**Issue by calvinmetcalf** [ngageoint/geopackage-js#116]
a promise based api would be pretty nice now that async await is available in most browers and node, any thoughts on switching to something like that in a newer version ?

**Comment by danielbarela** [comment]
I would not be opposed to doing something like that.  I think that is a good direction to take this library, since it is now available in most browsers and node.  I will add this to the list of things to do.

**Comment by stevage** [comment]
Came here to request the same thing. Well, a promise API specifically - probably still a bit premature to adopt async/await.

I tried using node-promisify but without success. For instance:

```
const promisify = require('promisify-node');
const geopackage = promisify('@ngageoint/geopackage');
geopackage.openGeoPackage('data.gpkg').then((gpkg) => {
    return promisify(gpkg.getFeatureTables)();
}).then(tables => {
    console.log(tables);
}).catch(console.error);;
```

caused 

```
TypeError: this.getGeometryColumnsDao is not a function
    at GeoPackage.getFeatureTables (/Users/stevebennett/odev/freelancing/crc-si/ziggurat/node_modules/@ngageoint/geopackage/lib/geoPackage.js:194:18)
```

I assume the `this` is not being maintained correctly.

**Comment by danielbarela** [comment]
I am actively working this issue on the api-changes branch

**Comment by danielbarela** [comment]
This has been published as 2.0.0
